File: projects/localserver/commons/src/main/java/com/robinfood/localserver/commons/entities/orders/OrderDetailProductEntity.java
package com.robinfood.localserver.commons.entities.orders;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@AllArgsConstructor
@Builder
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class OrderDetailProductEntity {

    private Long articleId;

    private Long articleTypeId;

    private Double basePrice;

    private Long brandId;

    private String brandName;

    private Long categoryId;

    private String categoryName;

    private BigDecimal co2Total;

    private BigDecimal deduction;

    private BigDecimal discount;

    private List<OrderDetailProductDiscountEntity> discounts;

    private Long finalProductId;

    private List<OrderDetailProductGroupEntity> groups;

    private Long id;

    private String image;

    private Long brandMenuId;

    private Long menuHallProductId;

    private String name;

    private Integer quantity;

    private Long sizeId;

    private String sizeName;

    private String sku;

    private List<OrderDetailProductTaxEntity> taxes;

    private Double unitPrice;

    private Double total;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/entities/OrderDetailProductEntity.kt
package com.robinfood.core.entities

import java.math.BigDecimal

data class OrderDetailProductEntity(
        val articleId: Long?,
        val articleName: String?,
        val articleTypeId: Long?,
        val basePrice: BigDecimal?,
        val brandMenuId: Long?,
        val brandName: String?,
        val categoryId: Long?,
        val categoryName: String?,
        val co2Total: BigDecimal?,
        val discounts: List<OrderDetailProductDiscountEntity>?,
        val displayType: Long?,
        val groups: List<OrderDetailProductGroupEntity>?,
        val id: Long?,
        val image: String?,
        val menuHallProductId: Long?,
        val name: String?,
        val quantity: Int?,
        val sizeId: Long?,
        val sizeName: String?,
        val sku: String?,
        val taxes: List<OrderDetailProductTaxEntity>?,
        val unitPrice: BigDecimal?
)
